src/lib/Toast/Toast.svelte

```diff
@@ -0,0 +1,154 @@
+<script lang="ts">
+  import { fly } from 'svelte/transition';
+
+  import { defaultToastProperties, type ToastProperties } from './properties';
+  import errorIcon from './error.svg';
+  import closeIcon from './close.svg';
+  import infoIcon from './info.svg';
+  import { createEventDispatcher, onDestroy, onMount } from 'svelte';
+
+  export let properties: ToastProperties = defaultToastProperties;
+  const dispatch = createEventDispatcher();
+
+  let visible = false;
+  let leftIcon = properties.leftIcon ?? null;
+  let timeout: ReturnType<typeof setInterval>;
+
+  properties = {
+    ...defaultToastProperties,
+    ...properties
+  };
+
+  async function hideToast() {
+    visible = false;
+  }
+
+  onMount(() => {
+    visible = true;
+    timeout = setInterval(hideToast, properties.duration);
+  });
+
+  onDestroy(() => {
+    clearInterval(timeout);
+  });
+
+  function getToastType(type: string): string | null {
+    switch (type) {
+      case 'success':
+        if (!leftIcon) {
+          leftIcon = 'https://sdk.breeze.in/gallery/icons/order-success-icon.svg';
+        }
+        return 'success-toast';
+      case 'error':
+        if (!leftIcon) {
+          leftIcon = errorIcon;
+        }
+        return 'error-toast';
+      case 'info':
+        if (!leftIcon) {
+          leftIcon = infoIcon;
+        }
+        return 'info-toast';
+      case 'warn':
+        if (!leftIcon) {
+          leftIcon = 'https://sdk.breeze.in/gallery/icons/warning.svg';
+        }
+        return 'warn-toast';
+      default:
+        return null;
+    }
+  }
+</script>
+
+{#if visible}
+  <div
+    class="toast {getToastType(properties.type)}"
+    style="border-radius: {properties.borderRadius};"
+    in:fly={{ x: 500, y: 0, duration: 400 }}
+    out:fly={{ x: 500, y: 0, duration: 800 }}
+    on:outroend={() => dispatch('kill')}
```

Comment:
move this logic to a function and dispatch from function itself

Reply:
updated.